Source/RimworldMod/Comp/ShipHeatMapComp.cs

```diff
@@ -268,7 +268,64 @@ public void SpawnEnemyShip(PassingShip passingShip)
             bool isDerelict = false;
             EnemyShipDef shipDef;
 
-            if (passingShip is AttackableShip)
+            //set ship faction
+            /*
+            Faction enemyshipfac = Faction.OfAncientsHostile;
+            if (shipDef.imperialShip && ModsConfig.RoyaltyActive && Faction.OfEmpire != null)
+            {
+                enemyshipfac = Faction.OfEmpire;
+                if (Faction.OfEmpire.AllyOrNeutralTo(Faction.OfPlayer))
+                    Faction.OfEmpire.TryAffectGoodwillWith(Faction.OfPlayer, -150);
+            }
+            else if (shipDef.mechanoidShip)
+                enemyshipfac = Faction.OfMechanoids;
+            else if (shipDef.pirateShip)
+                enemyshipfac = Faction.OfPirates;
+            */
+
+            // Find an enemy space navy
+            Faction enemyShipFaction = Faction.OfAncientsHostile;
+
+            IEnumerable<SpaceNavyDef> spaceNavyDefs = DefDatabase<SpaceNavyDef>.AllDefs;
+
+            SpaceNavyDef enemySpaceNavyDef = null;
+            EnemyShipDef enemySpaceNavyShipDef = null;
+
+            System.Random s_Random = new System.Random();
+
+            if (s_Random.Next(0, 100) <= ShipInteriorMod2.navyEncounterChance)
+            {
+                // Look through space navies
+                foreach (SpaceNavyDef spaceNavyDef in spaceNavyDefs)
+                {
+                    Faction spaceNavyFaction = Find.FactionManager.AllFactions.Where(faction => faction.def == spaceNavyDef.factionDef).First();
```

Comment:
Instead of .First(), use .RandomElement(), this way if more factions with the same def are present they can all be encountered.

```suggestion
                    Faction spaceNavyFaction = Find.FactionManager.AllFactions.Where(faction => faction.def == spaceNavyDef.factionDef).RandomElement();
```